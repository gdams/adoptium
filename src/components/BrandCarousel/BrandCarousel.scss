.slick-initialized .slick-track {
    display: flex;
    align-items: center;
}