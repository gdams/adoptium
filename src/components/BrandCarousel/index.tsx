import React from "react";
import Slider from "react-slick";

import "./BrandCarousel.scss"

const BrandCarousel = () => {
  const logos = [
    "microsoft.svg", // Replace with your logo image file names
    "ibm-logo.png",
    "adopters/logo-societe-generale.svg",
    "redhat.svg",
    "google.svg",
    "adopters/logo-dynatrace.png",
    "redhat.svg",
  ];

  const settings = {
    infinite: true,
    speed: 500,
    slidesToShow: 6,
    slidesToScroll: 1,
    autoplay: true,
    autoplaySpeed: 2000,
    pauseOnHover: true,
    responsive: [{
      breakpoint: 768,
      settings: {
        slidesToShow: 4
      }
    }, {
      breakpoint: 520,
      settings: {
        slidesToShow: 2
      }
    }]
  };

  return (
    <Slider {...settings}>
      {logos.map((logo, logoIndex) => (
        <div key={logoIndex}>
          <img
            src={`/images/${logo}`}
            className="mx-auto d-block"
            style={{
              filter: "grayscale(100%)",
              opacity: "0.7",
              maxWidth: "10em",
              transition: "filter 0.3s" 
            }}
            onMouseEnter={e => e.currentTarget.style.filter = "grayscale(0%)"}
            onMouseLeave={e => e.currentTarget.style.filter = "grayscale(100%)"}
            alt={`${logo} logo`}
          />
        </div>
      ))}
    </Slider>
  )
};

export default BrandCarousel;
