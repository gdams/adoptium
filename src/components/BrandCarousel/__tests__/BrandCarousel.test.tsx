import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest'
import BrandCarousel from '../index';

describe('BrandCarousel component', () => {
    it('should render correctly', () => {
        const { container } = render(
            <BrandCarousel />
        );

        expect(container).toMatchSnapshot();
    });
});
