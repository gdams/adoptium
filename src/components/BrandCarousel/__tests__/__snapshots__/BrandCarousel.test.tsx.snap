// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BrandCarousel component > should render correctly 1`] = `
<div>
  <div
    class="slick-slider slick-initialized"
    dir="ltr"
  >
    <button
      class="slick-arrow slick-prev"
      data-role="none"
      style="display: block;"
      type="button"
    >
       
      Previous
    </button>
    <div
      class="slick-list"
    >
      <div
        class="slick-track"
        style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
      >
        <div
          aria-hidden="true"
          class="slick-slide slick-cloned"
          data-index="-6"
          style="width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="ibm-logo.png logo"
                class="mx-auto d-block"
                src="/images/ibm-logo.png"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide slick-cloned"
          data-index="-5"
          style="width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="adopters/logo-societe-generale.svg logo"
                class="mx-auto d-block"
                src="/images/adopters/logo-societe-generale.svg"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide slick-cloned"
          data-index="-4"
          style="width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="redhat.svg logo"
                class="mx-auto d-block"
                src="/images/redhat.svg"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide slick-cloned"
          data-index="-3"
          style="width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="google.svg logo"
                class="mx-auto d-block"
                src="/images/google.svg"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide slick-cloned"
          data-index="-2"
          style="width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="adopters/logo-dynatrace.png logo"
                class="mx-auto d-block"
                src="/images/adopters/logo-dynatrace.png"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide slick-cloned"
          data-index="-1"
          style="width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="redhat.svg logo"
                class="mx-auto d-block"
                src="/images/redhat.svg"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="false"
          class="slick-slide slick-active slick-current"
          data-index="0"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="microsoft.svg logo"
                class="mx-auto d-block"
                src="/images/microsoft.svg"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="false"
          class="slick-slide slick-active"
          data-index="1"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="ibm-logo.png logo"
                class="mx-auto d-block"
                src="/images/ibm-logo.png"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="false"
          class="slick-slide slick-active"
          data-index="2"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="adopters/logo-societe-generale.svg logo"
                class="mx-auto d-block"
                src="/images/adopters/logo-societe-generale.svg"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="false"
          class="slick-slide slick-active"
          data-index="3"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="redhat.svg logo"
                class="mx-auto d-block"
                src="/images/redhat.svg"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="false"
          class="slick-slide slick-active"
          data-index="4"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="google.svg logo"
                class="mx-auto d-block"
                src="/images/google.svg"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="false"
          class="slick-slide slick-active"
          data-index="5"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="adopters/logo-dynatrace.png logo"
                class="mx-auto d-block"
                src="/images/adopters/logo-dynatrace.png"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide"
          data-index="6"
          style="outline: none; width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="redhat.svg logo"
                class="mx-auto d-block"
                src="/images/redhat.svg"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide slick-cloned"
          data-index="7"
          style="width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="microsoft.svg logo"
                class="mx-auto d-block"
                src="/images/microsoft.svg"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide slick-cloned"
          data-index="8"
          style="width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="ibm-logo.png logo"
                class="mx-auto d-block"
                src="/images/ibm-logo.png"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide slick-cloned"
          data-index="9"
          style="width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="adopters/logo-societe-generale.svg logo"
                class="mx-auto d-block"
                src="/images/adopters/logo-societe-generale.svg"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide slick-cloned"
          data-index="10"
          style="width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="redhat.svg logo"
                class="mx-auto d-block"
                src="/images/redhat.svg"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide slick-cloned"
          data-index="11"
          style="width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="google.svg logo"
                class="mx-auto d-block"
                src="/images/google.svg"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide slick-cloned"
          data-index="12"
          style="width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="adopters/logo-dynatrace.png logo"
                class="mx-auto d-block"
                src="/images/adopters/logo-dynatrace.png"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="slick-slide slick-cloned"
          data-index="13"
          style="width: 0px;"
          tabindex="-1"
        >
          <div>
            <div
              style="width: 100%; display: inline-block;"
              tabindex="-1"
            >
              <img
                alt="redhat.svg logo"
                class="mx-auto d-block"
                src="/images/redhat.svg"
                style="filter: grayscale(100%); opacity: 0.7; max-width: 10em; transition: filter 0.3s;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="slick-arrow slick-next"
      data-role="none"
      style="display: block;"
      type="button"
    >
       
      Next
    </button>
  </div>
</div>
`;
